webapp/src/components/Variants/Edition/DraggableCommands/CommandImportButton.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useRef } from 'react';
+import { ButtonBase, createStyles, makeStyles, Theme } from '@material-ui/core';
+import CloudUploadOutlinedIcon from '@material-ui/icons/CloudUploadOutlined';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    importIcon: {
+      width: '24px',
+      height: 'auto',
+      cursor: 'pointer',
+      color: theme.palette.primary.main,
+      margin: theme.spacing(0, 0.5),
+      '&:hover': {
+        color: theme.palette.secondary.main,
+      },
+    },
+    input: {
+      width: '200px',
+      marginLeft: theme.spacing(2),
+      marginRight: theme.spacing(2),
+    },
+  }));
+
+interface Inputs {
+    file: FileList;
+}
+
+interface PropTypes {
+  onImport: (json: object) => void;
+}
+
+const CommandImportButton = (props: PropTypes) => {
+  const { onImport } = props;
+  const classes = useStyles();
+  // const formRef = useRef(null);
+  // const { register, handleSubmit, getValues } = useForm<Inputs>();
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const onButtonClick = () => {
+    if (inputRef.current) {
+      inputRef.current.click();
+    }
+  };
+
+  const onInputClick = (e: React.MouseEvent<HTMLInputElement | MouseEvent>) => {
+    if (e && e.target) {
+      const element = e.target as HTMLInputElement;
+      element.value = '';
+    }
+  };
+
+  const handleChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    const reader = new FileReader();
+    reader.onload = async (ev: ProgressEvent<FileReader>) => {
+      if (ev.target) {
+        const text = (ev.target.result);
+        onImport(JSON.parse(text as string));
```

Comment:
Separate json parsing from onImport calling and add a try catch surrounding JSON.parse since it may fail and throw if the json is not properly formated.
Display an error box if it happens.